engine/action_contact.go

```diff
@@ -213,3 +216,55 @@ func (c *Client) GetContacts(ctx context.Context, metadata *Metadata, conditions
 
 	return contacts, nil
 }
+
+func (c *Client) AcceptContact(ctx context.Context, xPubID, paymail string) error {
+
+	contact, err := getContact(ctx, paymail, xPubID, c.DefaultModelOptions()...)
+	if err != nil {
+		c.Logger().Err(err).Msgf("unexpected error while geting contact for paymail %s and xPubID %s", paymail, xPubID)
+		return err
+	}
+	if contact == nil {
+		c.Logger().Err(ErrContactNotFound).Msgf("contact not found for paymail %s and xPubID %s", paymail, xPubID)
+		return ErrContactNotFound
+	}
+	if contact.Status != ContactAwaitAccept {
+		err = ErrContactStatusNotAwaiting
+		c.Logger().Err(err).Msgf("contact status is %s, expected %s", contact.Status, ContactAwaitAccept)
+		return err
+	}
+	contact.Status = ContactNotConfirmed
+	if err = contact.Save(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Client) RejectContact(ctx context.Context, xPubID, paymail string) error {
+
+	contact, err := getContact(ctx, paymail, xPubID, c.DefaultModelOptions()...)
+	if err != nil {
+		c.Logger().Err(err).Msgf("unexpected error while geting contact for paymail %s and xPubID %s", paymail, xPubID)
+		return err
+	}
+	if contact == nil {
+		c.Logger().Err(ErrContactNotFound).Msgf("contact not found for paymail %s and xPubID %s", paymail, xPubID)
+		return ErrContactNotFound
+	}
+	if contact.Status != ContactAwaitAccept {
+		err = ErrContactStatusNotAwaiting
```

Comment:
IMO it should be Warning

Reply:
Please add xpub and paymail to log context information

Reply:
changed to warning, xpub and paymail moved to context